speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -327,6 +377,128 @@ static void OnActivate(void* /* dummy */)
     LoadDefaultSettings();
 }
 
+
+/// @brief Releases a file dialog
+void ReleaseDialogAndCom(IFileDialog* fileDialog)
+{
+    fileDialog->Release();
+    CoUninitialize();
+}
+
+/// @brief Releases a shell item and a file dialog
+void ReleaseShellItemAndDialogAndCom(IShellItem* shellItem, IFileDialog* fileDialog)
+{
+    shellItem->Release();
+    ReleaseDialogAndCom(fileDialog);
+}
+
+/// @brief Edits the black box button with a message, releases a shell item and file dialog
+void ErrorBlackBoxButtonThenReleaseShellItemAndDialogAndCom(const char* message, IShellItem* shellItem, IFileDialog* fileDialog)
+{
+    m_blackBoxChosen = false;
+    GfuiButtonSetText(s_scrHandle, m_blackBoxButton, message);
+    ReleaseShellItemAndDialogAndCom(shellItem, fileDialog);
+}
```

Comment:
Maybe something like this might be clearer (it just uses parameter overloading to release more/less things).
It will be clear when calling the function because you are passing every object to be released as parameter.
```suggestion
/// @brief Releases a file dialog
void Release(IFileDialog* fileDialog)
{
    fileDialog->Release();
    CoUninitialize();
}

/// @brief Releases a shell item and a file dialog
void Release(IShellItem* shellItem, IFileDialog* fileDialog)
{
    shellItem->Release();
    Release(fileDialog);
}

/// @brief Edits the black box button with a message, releases a shell item and file dialog
void ShowErrorThenRelease(const char* errorMsg, IShellItem* shellItem, IFileDialog* fileDialog)
{
    m_blackBoxChosen = false;
    GfuiButtonSetText(s_scrHandle, m_blackBoxButton, errorMsg);
    Release(shellItem, fileDialog);
}
```